src/main/java/net/javadiscord/javabot/systems/moderation/AutoMod.java

```diff
@@ -121,25 +140,33 @@ private void checkContentAutomod(@Nonnull Message message) {
 					);
 			message.delete().queue();
 		}
+
 		final String messageRaw = message.getContentRaw();
+		Matcher urlMatcher = urlPattern.matcher(messageRaw);
 		if (messageRaw.contains("http://") || messageRaw.contains("https://")) {
 			// only do it for a links, so it won't iterate for each message
-			for (String spamUrl : spamUrls) {
-				if (messageRaw.contains(spamUrl)) {
-					try {
-						new ModerationService(message.getJDA(), Bot.config.get(message.getGuild()).getModeration())
-								.warn(
-										message.getMember(),
-										WarnSeverity.HIGH,
-										"Automod: Suspicious Link",
-										message.getGuild().getMember(message.getJDA().getSelfUser()),
-										message.getTextChannel(),
-										false
-								);
-						message.delete().queue();
-					} catch (Exception e) {
-						e.printStackTrace();
+			while (urlMatcher.find()) {
+				String url = messageRaw.substring(urlMatcher.start(1), urlMatcher.end());
+				try {
+					URI uri = new URI(url);
+					Bot.config.get(message.getGuild()).getModeration().getLogChannel().sendMessage(uri.getHost()).queue();
+					if (spamUrls.contains(uri.getHost())) {
+						if (message.getMember() != null){
+							Bot.config.get(message.getGuild()).getModeration().getLogChannel().sendMessage("Suspicious link by: ".concat("@" + message.getMember().getEffectiveName()).concat(" (" + message.getMember().getId() + ") ").concat(url)).queue();
```

Comment:
You could use `Misc.sendToLog(...);` instead

Reply:
done